sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosException.java

```diff
@@ -141,6 +142,11 @@ public class CosmosException extends AzureException {
      */
     private boolean sendingRequestHasStarted;
 
+    /***
+     * All selectable replica status.
+     */
+    private final List<String> replicaStatusList;
```

Comment:
the port will be different for each replica

did not start with including the full addresses because try not to limit the length of the diagnostics, but open for change

Reply:
Got it, where else do we print the replica information in diagnostics? In storePhysicalAddress field?

Reply:
Probably also contacted replicas, but none of them are logging all replicas information. The reason I want to log the health of all replicas is because it could help give us a better picture of why we choose this replica at that moment.